src/main/java/net/neoforged/neoforge/common/util/NbtListCollector.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.util;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.Tag;
+
+/**
+ * Stream support for converting a stream of NBT tags into an NBT ListTag.
+ * Useful for stream operations and mapping functions.
+ *
+ * Usage: {@code Stream.doStuff().collect(NbtListCollector.toNbtList())}
+ */
+public class NbtListCollector implements Collector<Tag, ListTag, ListTag> {
```

Comment:
Wouldn't this be more broadly applicable as a static method to build an `NbtList` from a list of tags? (I am surprised that this doesn't already exist, but couldn't find it).

Reply:
Maybe I'm interpreting badly, but there's a difference between doing a stream collection of `Tag`s into an `NbtList` (what this does) and turning a standard collection of `Tag`s into an `NbtList` (non-streaming)?

Reply:
After all, a list of `Tag` being put into a `ListTag` is simple code; a foreach will get you there no problem?